### PR TITLE
Improve global error handling

### PR DESCRIPTION
**Pull request by claudeatsafe**
Also started writing docs...

### CODE REVIEW

File: lib/shout_at.rb [hunk 1]
```diff
@@ -29,16 +29,31 @@ def init(opts, logger = nil)
...
     private
 
-    def method_builder(mod, accessor, level)
+    def method_builder(mod, accessor, level, rescue_handler)
       mod.define_singleton_method level do |message, opts = {}|
-        accessor.shout(message, opts)
+        begin
+          accessor.shout(message, opts)
+        rescue => e
+          rescue_handler.call(e)
```
Comment (pouellet) [inline lib/shout_at.rb]: Should we wrap the rescue handler in rescue block? I guess it's a bit meta

Comment (claudeatsafe) [inline lib/shout_at.rb]: Not quite sure what we would do in the rescue though, logging out?

Comment (pouellet) [inline lib/shout_at.rb]: I guess we could, but maybe we are better letting it bubble up.. If we ever screw up the Airbrake config we would basically never know about it. Not sure to be honest. Or we log and add an alert on logentries

Comment (claudeatsafe) [inline lib/shout_at.rb]: I let it bubble for now.

**Pull request merged**